### PR TITLE
feat: add maxLength to allow controlling max buffer length during decode

### PR DESCRIPTION
**Pull request by dryajov**
@dignifiedquire adding this since we need to control the max msg size in circuit. 

FYI, some tests are failing, but they were failing before my changes, I haven't yet taken a look at it, will do soon, but you might know whats going on.

### CODE REVIEW

**Comment by dryajov** [comment]
One more thing, I have tentatively set `maxLength` to 1024 as a default, but we might want to increase that to something larger if its too low of a number.

File: src/decode.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ exports.decodeFromReader = decodeFromReader
 
 const MSB = 0x80
 const isEndByte = (byte) => !(byte & MSB)
+const MAX_LENGTH = 1024
```
Comment (dignifiedquire) [inline src/decode.js]: go currently used 4mb as the max length, so lets make that the default

**Comment by dignifiedquire** [comment]
@dryajov I've updated master to have passing tests again. Please rebase onto it so I can merge

**Comment by dryajov** [comment]
@dignifiedquire thanks, I'll update the length!

**Comment by dryajov** [comment]
@dignifiedquire think this is ready now?

File: README.md [hunk 2]
```diff
@@ -70,6 +70,7 @@ Returns a pull-stream through.
 - `opts: Object`, optional
   - `fixed: false`:
   - `bytes: 4`: If `fixed` is `true` this is the amount of bytes used for the prefix.
+  - `maxLength`: If provided, will not decode messages longer than the size specified
```
Comment (dignifiedquire) [inline README.md]: this should mention the default length

File: test/index.spec.js [hunk 3]
```diff
@@ -52,6 +52,47 @@ describe('pull-length-prefixed', () => {
...
+      pull.values(input),
+      lp.encode(),
+      pull.collect((err, encoded) => {
+        if (err) throw err
+
+        expect(
+          encoded
+        ).to.be.eql([
+          Buffer.concat([
+            new Buffer(varint.encode('hello '.length)),
```
Comment (dignifiedquire) [inline test/index.spec.js]: please use `Buffer.from` instead

Comment (dignifiedquire) [inline test/index.spec.js]: or `Buffer.alloc(number)`

Comment (dryajov) [inline test/index.spec.js]: Should I just make the change everywhere in the tests while I'm at it?

Comment (dignifiedquire) [inline test/index.spec.js]: yeah, also that will require using `safe-buffer`, for node 4 compat

**Comment by dryajov** [comment]
@dignifiedquire any reason to not just do `require('safe-buffer')` and let it polyfill automatically here - https://github.com/dignifiedquire/pull-length-prefixed/blob/master/src/encode.js#L3?

**Comment by dryajov** [comment]
tests are failing on node 4 even tho I'm pulling `safe-buffer`

**Comment by dryajov** [comment]
Seems like in node 4, if the fill value is something other than a string and no encoding is explicitly specified, then it will not encode it with anything and simply fill in the raw bytes - https://github.com/nodejs/node/blob/v4.x/lib/buffer.js#L109, which is different from how later versions behave, where buffer is also defaulted to 'utf8'.

File: package.json [hunk 4]
```diff
@@ -32,6 +32,7 @@
   },
   "homepage": "https://github.com/dignifiedquire/pull-length-prefixed#readme",
   "dependencies": {
+    "mocha": "^3.2.0",
```
Comment (dignifiedquire) [inline package.json]: this shouldn't be needed

Comment (dryajov) [inline package.json]: Ah, thanks for catching that, leftovers from my tests figuring out what's wrong with node 4.

**Comment by dryajov** [comment]
can we get this published please 😄 ?

**Comment by dignifiedquire** [comment]
sorry for the delay, published a new version @dryajov

**Pull request merged**